### PR TITLE
[ci] Use Database interface instead of connection pool directly

### PR DESCRIPTION
**Pull request by daniel-goldstein**

### CODE REVIEW

**Comment by daniel-goldstein** [comment]
After the first commit, lots of unrelated type errors popped up from mypy. I think this is because I have the new mypy installed and it's actually catching more errors that were there all along. It might help to see the errors that it gave me (also visible in old CI builds of the PR:

```
ci/github.py:508: error: Incompatible types in assignment (expression has type "bool", variable has type "Optional[str]")
ci/github.py:551: error: Incompatible types in assignment (expression has type "bool", variable has type "Optional[str]")
ci/github.py:554: error: Incompatible types in assignment (expression has type "bool", variable has type "Optional[str]")
ci/github.py:574: error: Unsupported operand types for > ("int" and "None")
ci/github.py:574: note: Left operand is of type "Optional[int]"
ci/github.py:575: error: Unsupported operand types for + ("None" and "int")
ci/github.py:575: note: Left operand is of type "Optional[int]"
ci/github.py:817: error: Item "None" of "Optional[Dict[str, PR]]" has no attribute "values"
ci/github.py:828: error: Item "None" of "Optional[Dict[str, PR]]" has no attribute "values"
ci/github.py:840: error: Item "None" of "Optional[Dict[str, PR]]" has no attribute "values"
ci/github.py:842: error: Item "None" of "Optional[Dict[str, PR]]" has no attribute "values"
ci/github.py:849: error: Item "MergeFailureBatch" of "Union[Batch, Any, MergeFailureBatch]" has no attribute "id"
ci/github.py:849: error: Item "None" of "Optional[Dict[str, PR]]" has no attribute "values"
Found 11 errors in 1 file (checked 19 source files)
```

It might be helpful to look at the first commit and last commit in isolation. Or if you'd like I can make a separate PR.

File: ci/ci/ci.py [hunk 1]
```diff
@@ -368,7 +364,7 @@ async def push_callback(event):
         branch_name = ref[len('refs/heads/') :]
         branch = FQBranch(Repo.from_gh_json(data['repository']), branch_name)
         for wb in watched_branches:
-            if wb.branch == branch or any(pr.branch == branch for pr in wb.prs.values()):
```
Comment (danking) [inline ci/ci/ci.py:371]: huh, I don't understand how this didn't cause attribute errors.

File: ci/ci/github.py [hunk 2]
```diff
@@ -628,7 +623,7 @@ def __init__(self, index, branch, deployable, mergeable):
         self.deployable: bool = deployable
         self.mergeable: bool = mergeable
 
-        self.prs: Optional[Dict[str, PR]] = None
+        self.prs: Dict[str, PR] = {}
```
Comment (danking) [inline ci/ci/github.py:626]: huh this was silly.

File: ci/ci/github.py [hunk 3]
```diff
@@ -640,7 +635,7 @@ def __init__(self, index, branch, deployable, mergeable):
         self.batch_changed: bool = True
         self.state_changed: bool = True
 
-        self.n_running_batches: Optional[int] = None
+        self.n_running_batches: int = 0
```
Comment (danking) [inline ci/ci/github.py:638]: I think the idea was that None meant we had no information yet. For example, we maybe shouldn't run any PRs until we know how many running batches we have? I'm, OK with this change too if its simpler.

**Review by danking — APPROVED** [review]

**Pull request merged**